evalml/automl/automl_search.py

```diff
@@ -857,16 +858,34 @@ def _handle_keyboard_interrupt(self):
             else:
                 leading_char = ""
 
-    def search(self, show_iteration_plot=True):
+    def search(self, show_iteration_plot=True, timing=None):
         """Find the best pipeline for the data set.
 
         Args:
             show_iteration_plot (boolean, True): Shows an iteration vs. score plot in Jupyter notebook.
                 Disabled by default in non-Jupyter enviroments.
+            timing (str, None): Shows timing of the batches and the individual timings of each pipeline.
+                Default: None
+                log=prints out batch/pipeline timing to console.
 
         Raises:
             AutoMLSearchException: If all pipelines in the current AutoML batch produced a score of np.nan on the primary objective.
+            ValueError: If timing is not set to a correct value
+
+        Returns:
+                Dict[int, Dict[str, Timestamp]]: Returns dict.
+                Key=batch #, value=Dict[key=pipeline name, value=timestamp of pipeline].
+                Inner dict has key called "Total time of batch" with value=total time of batch.
```

Comment:
This is really hard to understand without reading closely. I'd refactor it, something more like:
```
Dict[int, Dict[str, Timestamp]]: Dictionary keyed by batch number that maps to the timings for pipelines run in that batch, 
as well as the total time for each batch. Pipelines within a batch are labeled by pipeline name. 
```

As a side note, `=` in docstrings really throws me off. It'd be better to stick to using colons, which maintain consistency with the rest of our docs!